common/functions/src/scalars/arithmetics/arithmetic_mul.rs

```diff
@@ -33,8 +34,154 @@ impl ArithmeticMulFunction {
+        arithmetic_mul_div_monotonicity(args, DataValueArithmeticOperator::Mul)
+    }
+}
+
+pub fn arithmetic_mul_div_monotonicity(
+    args: &[Monotonicity],
+    op: DataValueArithmeticOperator,
+) -> Result<Monotonicity> {
+    if args.len() != 2 {
+        return Err(ErrorCode::BadArguments(format!(
+            "Invalid argument lengths {} for get_monotonicity",
+            args.len()
+        )));
+    }
+
+    let err_opt = if let DataValueArithmeticOperator::Mul | DataValueArithmeticOperator::Div = op {
+        None
+    } else {
+        Some(Err(ErrorCode::BadArguments(format!(
+            "Invalid operator '{}' for get_monotonicity",
+            op
+        ))))
+    };
+    if let Some(err) = err_opt {
+        return err;
+    }
+
+    let flip_positive_by_operator = |positive: bool| -> bool {
+        match op {
+            DataValueArithmeticOperator::Mul => positive,
+            DataValueArithmeticOperator::Div => !positive,
+            _ => unreachable!(),
+        }
+    };
+
+    let f_x = &args[0];
+    let g_x = &args[1];
+
+    match (f_x.is_constant, g_x.is_constant) {
+        // both f(x) and g(x) are constant
+        (true, true) => Ok(Monotonicity::create_constant()),
+
+        //f(x) is constant
+        (true, false) => {
+            if f_x.gt_eq_zero()? {
+                // 12 *|/ g(x)
+                Ok(Monotonicity::create(
+                    g_x.is_monotonic,
+                    flip_positive_by_operator(g_x.is_positive),
+                    g_x.is_constant,
+                ))
+            } else if f_x.lt_eq_zero()? {
+                // -12 */ g(x)
+                Ok(Monotonicity::create(
+                    g_x.is_monotonic,
+                    !flip_positive_by_operator(g_x.is_positive),
+                    g_x.is_constant,
+                ))
+            } else {
+                // unknown monotonicity
+                Ok(Monotonicity::default())
+            }
+        }
+
+        // g(x) is constant
+        (false, true) => {
+            if g_x.gt_eq_zero()? {
+                // f(x) *|/ 12
+                Ok(Monotonicity::create(
+                    f_x.is_monotonic,
+                    f_x.is_positive,
+                    f_x.is_constant,
+                ))
+            } else if g_x.lt_eq_zero()? {
+                // f(x) *|/ (-12), need to flip the is_positive for negative constant value.
+                Ok(Monotonicity::create(
+                    f_x.is_monotonic,
+                    !f_x.is_positive,
+                    f_x.is_constant,
+                ))
+            } else {
+                // unknown monotonicity
+                Ok(Monotonicity::default())
+            }
+        }
+
+        // neither f(x) nor g(x) are constant, like abs(x) *|- (x+12)
+        (false, false) => {
+            if f_x.is_monotonic && g_x.is_monotonic {
+                let f_x_gt_eq_zero = f_x.gt_eq_zero()?;
```

Comment:
Maybe we can create a function `fn max(left, right)-> Result<i8>`, 1: left>right, 0: left=right, -1 left<right